### ISSUE TITLE
Embedding package files through the project properties UI doesn't work when the package files are in the same directory as the project file

### ISSUE THREAD

**Issue by chgill-MSFT** [dotnet/project-system#7642]
## Visual Studio Version

I am using Visual Studio 2022 preview 5, however I believe the bug repros for all VS 2022 previews.

## Summary

Embedding package files (README, package icon, license) through the package properties UI is broken when the files are in the same directory as the project file. 

Embedding package elements through the package properties UI works as expected when the files are outside the project file directory.

## Steps to Reproduce

1. Create a class library project in VS 2022.
2. Add a README.md file to the same directory as the project file.
3. Add the README.md file to the package through the project properties UI. This generates the following XML in the project file:

`<PackageReadmeFile>README.md</PackageReadmeFile>`

and

```xml
  <ItemGroup>
    <None Update="README.md">
      <Pack>True</Pack>
    </None>
  </ItemGroup>
```

4. Attempt to pack the package
5. Pack fails with the following error even though the file is there:

`NU5046: The readme file 'README.md' does not exist in the package.`

**Video from @timheuer:**

![2021-09-24_14-59-59](https://user-images.githubusercontent.com/4821/134744065-77d8dbee-eb9a-4961-b0a4-9f1bdac43b02.gif)

## Expected Behavior

The generated XML should be:

`<PackageReadmeFile>README.md</PackageReadmeFile>`

and

```xml
  <ItemGroup>
    <None Include="README.md">
      <Pack>True</Pack>
      <PackagePath></PackagePath>
    </None>
  </ItemGroup>
```

This is already the behavior when embedded files are outside the project file directory.

## Actual Behavior

When the package files are in the project file directory, the generated XML changes from `<None Include="README.md">` into `<None Update="README.md">` and omits the `<PackagePath>` node. This causes pack to fail.

`<PackageReadmeFile>README.md</PackageReadmeFile>`

and

```xml
  <ItemGroup>
    <None Update="README.md">
      <Pack>True</Pack>
    </None>
  </ItemGroup>
```

Another example using icons in the following image comparing VS 16.11 and VS 17.0 preview 5

<img src="https://user-images.githubusercontent.com/15097183/134979491-b35bab4b-974a-46e6-869c-88612146f5dc.png" width="500">

## User Impact

Embedding package files when they are in the same directory as the project file is broken in VS 2022. This is a very common scenario, especially for README files since each package in a repo will likely have a package-specific README in it's project file directory.

This is likely to lead to significant frustration for package authors using VS 2022.

## Addition examples for different package file locations

**README is in same directory as proj file**
<img src="https://user-images.githubusercontent.com/15097183/134973019-96797d70-a05f-4fc6-a4b8-7cc58a44f0da.png" width="500">

**README is in solution root directory**
<img src="https://user-images.githubusercontent.com/15097183/134974136-94949a61-b99d-45dc-b0f9-40334bfb6539.png" width="500">

**README is outside repo solution root directory**
<img src="https://user-images.githubusercontent.com/15097183/134974320-d97001e4-8d2d-480d-9f0a-96fee6c2931b.png" width="500">

**Comment by MiYanni** [comment]
@chgill-MSFT Is an empty item attribute significant? Meaning, does `<PackagePath></PackagePath>` cause something different to happen versus it not being there? Since we're used to working with MSBuild loading a .csproj file and treating both an empty item attribute versus the attribute not being there at all as the same thing, is the packing process evaluating the file differently?

**Comment by chgill-MSFT** [comment]
@zkat might have a better understanding of why the empty `<PackagePath><PackagePath>` is necessary. I can confirm that pack doesn't work without the PackagePath element even if "Update" is changed to "Include".

**Comment by zkat** [comment]
Hi @MiYanni

We interpret a missing `PackagePath` as wanting to put stuff into `content`/`contentFiles`:
![image](https://user-images.githubusercontent.com/17535/135663547-249d769b-b776-4111-83ac-154d87f03a53.png)

Whereas an empty `PackagePath` is treated as package root:
![image](https://user-images.githubusercontent.com/17535/135663912-0aba64bf-ebbe-4720-ae70-bd369a6cdf1f.png)

**Comment by MiYanni** [comment]
@zkat That generally goes against the spirit of the file. It might make sense from a nuspec perspective, but not from an MSBuild/XML perspective. I'll add the empty `PackagePath` node, but it will look odd to anyone who knows generally how this file (.csproj) works. Additionally, the documentation for these items doesn't mention this information about `PackagePath` defaulting to `content/contentFiles`. I primarily looked at this page here: https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packagereadmefile

**Comment by zkat** [comment]
I don't think this is something we can change? As far as I know, it's always been like this, and we have really strong backwards-compat guarantees, so we even have to keep the smells. /cc @nkolev92, @zivkan, @jeffk for further thoughts? Maybe there's something we could do?

**Comment by MiYanni** [comment]
@zkat To avoid the smell, I could do `<PackagePath>"\"</PackagePath>` instead. At least it is explicit. That should be equivalent to the empty value, correct?

**Comment by zkat** [comment]
I think that's the same? It ends up with a different target (`\readme.md` vs `readme.md`) but I'm pretty sure those are the same? I think it's good enough, at least. :)

**Comment by nkolev92** [comment]
> That generally goes against the spirit of the file. It might make sense from a nuspec perspective, but not from an MSBuild/XML perspective.

Can you elaborate why a PackagePath doesn't make sense from MSBuild perspective? 

fwiw, the page you linked, https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packagereadmefile, does include a PackagePath explicitly.

**Comment by zivkan** [comment]
> Can you elaborate why a PackagePath doesn't make sense from MSBuild perspective?

I believe it's referring to an empty value being different to the value not being defined. Given the way you implement default values in MSBuild is with `Condition=" '$(MyVar)' == '' "`, I think it's a clear signal that in general empty and undefined are more or less the same thing. 

Having said that, I don't see how we can change it without breaking everyone who has been using it so far.

**Comment by MiYanni** [comment]
@nkolev92 The page linked does not mention **exactly** what I said, that `<PackagePath></PackagePath>` is equivalent to `<PackagePath>content;contentFiles</PackagePath>`. There are 2 lines that mention something similar, but this is explicitly described for `ContentTargetFolders`, which is a commandline input for the `pack` operation.

![image](https://user-images.githubusercontent.com/17788297/135694717-176a3179-9179-426f-937e-e1c790812852.png)

![image](https://user-images.githubusercontent.com/17788297/135694756-f855c7f7-71a4-4b16-bfd0-2dcfe045fe22.png)

Why `<PackagePath></PackagePath>` doesn't make sense is that the absence of an MSBuild element is equivalent to empty string. When you do `$(PackagePath)`, it will return empty string in both if the value was not defined (unset) or if it was defined as `<PackagePath></PackagePath>`. Both are equal. @zivkan, thanks for helping clarify.

**Comment by nkolev92** [comment]
> I believe it's referring to an empty value being different to the value not being defined. Given the way you implement default values in MSBuild is with Condition=" '$(MyVar)' == '' ", I think it's a clear signal that in general empty and undefined are more or less the same thing.

Oh, I totally missed that part.

That makes sense @MiYanni, I thought you were referring to something different.

Btw, I don't think the first example in `Addition examples for different package file locations` in the body seems incorrect.
README is in same directory as proj file

PackagePath is about the output, where the input is coming from is *irrelevant*. 

`\` is better.

**Comment by nkolev92** [comment]
Bug probably here, https://github.com/NuGet/NuGet.Client/blob/966b1b21eec28834365a8604744dab682c8326ea/src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs#L630, but yeah, I'm not sure we can change it easily. 

We can add a warning and fix it later.

**Comment by zkat** [comment]
A warning is unfortunately still a breaking change for many of our customers :(

**Comment by MiYanni** [comment]
Got some time to investigate this. It isn't my code that is causing the issue. It seems that the mechanism we use to add these elements is 'smart' and does different things depending on the situation. Here, if any file that is selected is at the project's directory or lower in the directory structure, it is already in the project because of globbing. That's why it changes from a `None` item into an `Update` item. However, there is clearly some other 'cleanup' type logic that happens because it sees an empty node and removes it. Again, this reinforces that empty value nodes are not significant in the MSBuild world. If I simply change the default from `string.Empty` to `"\\"`, it fixes the issue since I now have a value to provide it and it doesn't get optimized away. So, I'm going to go with that change. I'd have to look deeper to why that optimization happens, but it isn't relevant at the moment to fix this issue.
